python/paddle/tensor/manipulation.py

```diff
@@ -794,7 +794,8 @@ def gather(x, index, axis=None, name=None):
         axis = 0
     axis_tensor = axis
     if not isinstance(axis, Variable):
-        axis_tensor = fill_constant(shape=[1], dtype='int64', value=axis)
```

Comment:
No need to cast axis to axis_tensor and it may introduce extra performance loss. 
gather op supports `axis` as an attribute.